### PR TITLE
[components] Minor tweaks to make the demo flow work

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Went through some rounds of iteration on getting the demo flow to work, uncovering some bugs along the way

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
* **#26338** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26338?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26338?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/libraries/dagster-components/dagster_components/core/component_defs_builder.py [hunk 1]
```diff
@@ -127,7 +127,11 @@ def build_defs_from_toplevel_components_folder(
     """Build a Definitions object from an entire component hierarchy."""
     from dagster._core.definitions.definitions_class import Definitions
 
-    context = CodeLocationProjectContext.from_path(path, registry or ComponentRegistry.empty())
+    from dagster_components import __component_registry__
+
+    context = CodeLocationProjectContext.from_path(
+        path, registry or ComponentRegistry(__component_registry__)
+    )
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-components/dagster_components/core/component_defs_builder.py:134]: note: was ok with this being gross in the short term as sean's PR will expose the "correct" way of doing this

File: python_modules/libraries/dagster-components/dagster_components/impls/dbt_project.py [hunk 2]
```diff
@@ -109,11 +109,14 @@ def _fn(context: AssetExecutionContext, dbt: DbtCliResource):
...
     def generate_files(cls, params: DbtGenerateParams) -> Mapping[str, Any]:
+        cwd = os.getcwd()
         if params.project_path:
-            relative_path = os.path.relpath(params.project_path, start=os.getcwd())
+            relative_path = os.path.relpath(params.project_path, start=cwd)
         elif params.init:
             dbtRunner().invoke(["init"])
-            subpaths = list(Path(os.getcwd()).iterdir())
+            subpaths = [
+                path for path in Path(cwd).iterdir() if path.is_dir() and path.name != "logs"
```
Comment (schrockn) [inline python_modules/libraries/dagster-components/dagster_components/impls/dbt_project.py:118]: What's up with the "logs" special-casing?

Comment (OwenKephart) [reply python_modules/libraries/dagster-components/dagster_components/impls/dbt_project.py:118]: dbt generates a `logs` directory to store logs when executing any command (including dbt init). Realistically should delete that dir here instead, just ignored it for now

**Review by schrockn — APPROVED** [review]

**Pull request merged**